ai21/clients/common/beta/assistant/plans.py

```diff
@@ -15,20 +19,33 @@ def create(
         self,
         *,
         assistant_id: str,
-        code: str,
+        code: str | callable,
+        schemas: list[dict] | list[Type[BaseModel]] | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> PlanResponse:
         pass
 
     def _create_body(
         self,
         *,
-        code: str,
+        code: str | callable,
+        schemas: list[dict] | list[Type[BaseModel]] | NotGiven = NOT_GIVEN,
         **kwargs,
     ) -> Dict[str, Any]:
+        if callable(code):
+            code = inspect.getsource(code).strip()
```

Comment:
I think we should wrap it in try except in case the parsing fails so we can throw an appropriate error. 
For instance, we the user is running our code from the terminal and would run something like this - 

```python
def foo():
    pass

AI21Client().beta.assistant.plan.create(code=foo)
```
They will get the following error - 
```shell
, line 1096, in findsource
    raise OSError('could not get source code')
OSError: could not get source code
```

Which will be unclear. So raising a custom error is essential. You can use the errors we have [here](https://github.com/AI21Labs/ai21-python/blob/main/ai21/errors.py) or create your own :)